Nef_3/include/CGAL/Nef_3/SNC_decorator.h

```diff
@@ -851,146 +700,14 @@ class SNC_decorator : public SNC_const_decorator<Map> {
     return c->shells_end();
   }
 
-  Size_type number_of_vertices() const
-  { return sncp()->number_of_vertices(); }
-  Size_type number_of_halfedges() const
-  { return sncp()->number_of_halfedges(); }
-  Size_type number_of_edges() const
-  { return sncp()->number_of_edges(); }
-  Size_type number_of_halffacets() const
-  { return sncp()->number_of_halffacets();}
-  Size_type number_of_facets() const
-  { return sncp()->number_of_facets();}
-  Size_type number_of_volumes() const
-  { return sncp()->number_of_volumes();}
+  using Base::number_of_vertices;
+  using Base::number_of_halfedges;
+  using Base::number_of_edges;
+  using Base::number_of_halffacets;
+  using Base::number_of_facets;
+  using Base::number_of_volumes;
 
 };
 
-/* visiting shell objects:
-
-Objects are marked as done, when placed in the output list.  We have
-to maintain a stack of sface candidates (the spherical rubber sectors
-that provide connectivity at the local graphs of vertices) and facet
-candiates (the plane pieces in three space also providing
-connectivity). Note that we have to take care about the orientation of
-sobjects and facets. We have to take care that (1) the search along
-the shell extends along the whole shell structure (2) does not visit
-any object twice, and (3) all 3-space objects have to be reported and
-this also just once.
-
-The facets and sfaces are marked |done| when they are put into their
-corresponding queues thus each such object is visited exactly once
-when taken out of the queue.
-
-When an sface |sf| is taken out of the queue |SFaceCandiates| its
-boundary structure is examined and all 2-skeleton objects (vertices
-and edges of 3-space) that are incident to the volume represented by
-|sf| are reported. Facets are reported when they are taken out of
-|FacetCandiates|.
-
-*/
-
-template <typename EW>
-template <typename Visitor>
-void SNC_decorator<EW>::
-visit_shell_objects(SFace_handle f, Visitor& V) const
-{
-  typedef typename SM_decorator::SHalfedge_around_sface_circulator
-    SHalfedge_around_sface_circulator;
-  std::list<SFace_handle> SFaceCandidates;
-  std::list<Halffacet_handle> FacetCandidates;
-  CGAL::Generic_handle_map<bool> Done(false);
```

Comment:
@afabri It's not a "sneak in", this is how it was implemented in SNC_decorator. It does however require the void cast in Generic_handle_map.